bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -1343,38 +1393,39 @@ synchronized public void waitZero() throws InterruptedException {
      */
     public static boolean format(ServerConfiguration conf,
             boolean isInteractive, boolean force) {
-        File journalDir = conf.getJournalDir();
-        String[] journalDirFiles =
-                journalDir.exists() && journalDir.isDirectory() ? journalDir.list() : null;
-        if (journalDirFiles != null && journalDirFiles.length != 0) {
-            try {
-                boolean confirm = false;
-                if (!isInteractive) {
-                    // If non interactive and force is set, then delete old
-                    // data.
-                    if (force) {
-                        confirm = true;
+
+        boolean alreadyConfirmed = false;
+        for (File journalDir : conf.getJournalDirs()) {
+            if (journalDir.exists() && journalDir.isDirectory() && journalDir.list().length != 0) {
+                try {
+                    boolean confirm = false;
+                    if (!isInteractive) {
+                        // If non interactive and force is set, then delete old
+                        // data.
+                        if (force) {
+                            confirm = true;
+                        } else {
+                            confirm = false;
+                        }
                     } else {
-                        confirm = false;
+                        confirm = alreadyConfirmed || IOUtils.confirmPrompt("Are you sure to format Bookie data..?");
+                        alreadyConfirmed = true;
```

Comment:
This logic is confusing. All you need to do is, move confirm outside the for loop and in the else case, if (!confirmed) IOUtils... will do the job. You don't need alreadyConfirmed flag.